desktop/extensions-sdk/guides/invoke-host-binaries.md

```diff
@@ -0,0 +1,162 @@
+```
+
+Then update the `Dockerfile` to copy the `binaries` folder into the extension container.
+
+```dockerfile
+# Copy the binaries into the right folder
+COPY binaries/windows/hello.ps1 /windows/hello.ps1
+COPY binaries/unix/hello.sh /linux/hello.sh
+COPY binaries/unix/hello.sh /darwin/hello.sh
+```
+
+## Invoke the host binary from the UI
+
+In your app, use the Docker Desktop Client object and then invoke the binary provided by the extension with the 
+`ddClient.extension.host.cli.exec()` function.
+In this example, the binary returns a string as result, obtained by `result?.stdout`, as soon as the app starts.
+
+<ul class="nav nav-tabs">
+  <li class="active"><a data-toggle="tab" data-target="#react-app" data-group="react">For React</a></li>
+  <li><a data-toggle="tab" data-target="#vue-app" data-group="vue">For Vue</a></li>
+  <li><a data-toggle="tab" data-target="#angular-app" data-group="angular">For Angular</a></li>
+  <li><a data-toggle="tab" data-target="#svelte-app" data-group="svelte">For Svelte</a></li>
+</ul>
+
+<div class="tab-content">
+  <div id="react-dockerfile" class="tab-pane fade in active" markdown="1">
+
+```typescript
+
+export function App() {
+  const ddClient = createDockerDesktopClient();
+  const [hello, setHello] = useState("");
+
+  useEffect(() => {
+    const run = async () => {
+      let binary = "hello.sh";
+      if (ddClient.host.platform === 'win32') {
+        binary = "hello.ps1";
+      }
+
+      const result = await ddClient.extension.host?.cli.exec(binary, ["world"]);
+      setHello(result?.stdout);
+
+    };
+    run();
+  }, [ddClient]);
+    
+  return (
+    <div>
+      {hello}
+    </div>
+  );
+}
+```
+
+
+  </div>
+  <div id="vue-app" class="tab-pane fade" markdown="1">
+
+<br/>
+
+> **Important**
+> We don't have an example for Vue yet. [Fill out the form](https://docs.google.com/forms/d/e/1FAIpQLSdxJDGFJl5oJ06rG7uqtw1rsSBZpUhv_s9HHtw80cytkh2X-Q/viewform?usp=pp_url&entry.1333218187=Vue)
+> and let us know you'd like a sample with Vue.
+{: .important }
+
+  </div>
+  <div id="angular-app" class="tab-pane fade" markdown="1">
+
+<br/>
+
+> **Important**
+> We don't have an example for Angular yet. [Fill out the form](https://docs.google.com/forms/d/e/1FAIpQLSdxJDGFJl5oJ06rG7uqtw1rsSBZpUhv_s9HHtw80cytkh2X-Q/viewform?usp=pp_url&entry.1333218187=Angular)
+> and let us know you'd like a sample with Angular.
+{: .important }
+
+  </div>
+  <div id="svelte-app" class="tab-pane fade" markdown="1">
+
+<br/>
+
+> **Important**
+> We don't have an example for Svelte yet. [Fill out the form](https://docs.google.com/forms/d/e/1FAIpQLSdxJDGFJl5oJ06rG7uqtw1rsSBZpUhv_s9HHtw80cytkh2X-Q/viewform?usp=pp_url&entry.1333218187=Svelte)
+> and let us know you'd like a sample with Svelte.
+{: .important }
+
+  </div>
+</div>
+
+## Configure the metadata file
+
+The host binaries must be specified in the `metadata.json` so that Docker Desktop copies them on the host when installing the extension.
```

Comment:
```suggestion
The host binaries must be specified in the `metadata.json` so that Docker Desktop copies them on the host when installing the extension. Once the extension is uninstalled, the binaries that were copied will be removed as well.
```